Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectDiffScan.py

```diff
@@ -0,0 +1,278 @@
+from mantid.kernel import *
+from mantid.api import *
+from mantid.simpleapi import *
+from mantid import config
+
+
+class IndirectDiffScan(DataProcessorAlgorithm):
+    _data_files = None
+    _sum_files = None
+    _load_logs = None
+    _calibration_ws = None
+    _instrument_name = None
+    _analyser = None
+    _reflection = None
+    _efixed = None
+    _spectra_range = None
+    _background_range = None
+    _elastic_range = None
+    _inelastic_range = None
+    _rebin_string = None
+    _detailed_balance = None
+    _grouping_method = None
+    _grouping_ws = None
+    _grouping_map_file = None
+    _output_ws = None
+    _output_x_units = None
+    _plot_type = None
+    _save_formats = None
+    _ipf_filename = None
+    _sample_log_name = None
+    _sample_log_value = None
+    _workspace_names = None
+    _scan_ws = None
+
+    def category(self):
+        return 'Workflow\\Inelastic;Inelastic\\Indirect;Workflow\\MIDAS'
+
+    def summary(self):
+        return 'Runs an energy transfer reduction for an inelastic indirect geometry instrument.'
+
+    def PyInit(self):
+        # Input properties
+        self.declareProperty(StringArrayProperty(name='InputFiles'),
+                             doc='Comma separated list of input files')
+
+        self.declareProperty(name='LoadLogFiles', defaultValue=True,
+                             doc='Load log files when loading runs')
+
+        # Instrument configuration properties
+        self.declareProperty(name='Instrument', defaultValue='',
+                             validator=StringListValidator(['IRIS', 'OSIRIS']),
```

Comment:
It gets passed to `ISISIndirectDiffractionReduction` where it is used for the instrument parameter file. It could be an idea to refactor `_get_InstrRun()` to just `_get_run()`